azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/maven/MavenSettingHelper.java

```diff
@@ -0,0 +1,101 @@
+        if (configuration == null) {
+            return configurationFromServer;
+        }
+        configurationFromServer.setTenant(getConfiguration(configuration, "tenant"));
+        configurationFromServer.setClient(getConfiguration(configuration, "client"));
+        final String rawKey = getConfiguration(configuration, "key");
+        configurationFromServer.setKey(isValueEncrypted(rawKey) ? decryptMavenProtectedValue(settingsDecrypter, "key", rawKey) : rawKey);
+        configurationFromServer.setCertificate(getConfiguration(configuration, "certificate"));
+        final String rawCertificatePassword = getConfiguration(configuration, "certificatePassword");
+        configurationFromServer.setCertificatePassword(isValueEncrypted(rawCertificatePassword) ?
+                decryptMavenProtectedValue(settingsDecrypter, "certificatePassword", rawCertificatePassword) :
+                rawCertificatePassword);
+        configurationFromServer.setEnvironment(getConfiguration(configuration, "environment"));
+        configurationFromServer.setHttpProxyHost(getConfiguration(configuration, "httpProxyHost"));
+        configurationFromServer.setHttpProxyPort(getConfiguration(configuration, "httpProxyPort"));
+        configurationFromServer.setServerId(serverId);
+
+        // validate configuration
+        return configurationFromServer.validateAndReturn();
+    }
+
+    /**
+     * Get string value from server configuration section in settings.xml.
+     *
+     * @param configuration the <code>Xpp3Dom</code> object representing the configuration in &lt;server&gt;.
+     * @param property           the property name
+     * @return String value if property exists; otherwise, return null.
+     */
+    private static String getConfiguration(final Xpp3Dom configuration, final String property) {
+        final Xpp3Dom node = configuration.getChild(property);
+        return Objects.isNull(node) ? null : node.getValue();
+    }
+
+    private static boolean isValueEncrypted(String value) {
+        return value != null && value.startsWith("{") && value.endsWith("}");
+    }
+
+    private static String decryptMavenProtectedValue(SettingsDecrypter settingsDecrypter, String propertyName, String value) throws MavenDecryptException {
+        final Server server = new Server();
+        server.setPassword(value);
+        final SettingsDecryptionRequest request = new DefaultSettingsDecryptionRequest(server);
+        final SettingsDecryptionResult result = settingsDecrypter.decrypt(request);
+        for (final SettingsProblem problem : result.getProblems()) {
+            if (problem.getSeverity() == SettingsProblem.Severity.ERROR || problem.getSeverity() == SettingsProblem.Severity.FATAL) {
+                // for java 8+, it is ok to use operator '+' for string concatenation
+                throw new MavenDecryptException(propertyName, value, "Unable to decrypt value(" + value + ") from settings.xml: " + problem);
```

Comment:
I prefer to user SettingsProblem.Severity.ERROR.equals(problem.getSeverity()) ...

Reply:
enum prefers ==

Reply:
https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals